jsonstream.cpp warning regarding an unused variable under ARM

```
src/jsonstream.cpp:194:14: warning: unused variable ‘supports’ [-Wunused-variable]
     uint32_t supports = detect_supported_architectures();
              ^~~~~~~~
```

--- a/include/simdjson/isadetection.h
+++ b/include/simdjson/isadetection.h
@@ -74,7 +74,7 @@ enum instruction_set {
 
 #if defined(__arm__) || defined(__aarch64__) // incl. armel, armhf, arm64
 
-#if defined(__NEON__)
+#if defined(__ARM_NEON)
 
 static inline uint32_t detect_supported_architectures() {
   return instruction_set::NEON;

--- a/src/jsonparser.cpp
+++ b/src/jsonparser.cpp
@@ -42,7 +42,7 @@ Architecture find_best_supported_implementation() {
     return Architecture::HASWELL;
   if ((westmere_flags & supports) == westmere_flags)
     return Architecture::WESTMERE;
-  if (instruction_set::NEON)
+  if (supports & instruction_set::NEON)
     return Architecture::ARM64;
 
   return Architecture::NONE;

--- a/src/jsonstream.cpp
+++ b/src/jsonstream.cpp
@@ -211,7 +211,7 @@ void find_the_best_supported_implementation() {
     }
 #endif
 #ifdef IS_ARM64
-    if (instruction_set::NEON) {
+    if (supports & instruction_set::NEON) {
         best_stage1 = simdjson::find_structural_bits<Architecture ::ARM64>;
         best_stage2 = simdjson::unified_machine<Architecture ::ARM64>;
         return;

